calicoctl.py

```diff
@@ -490,12 +470,111 @@ def group_show(detailed):
                 x.add_row([name, "No members"])
     else:
         x = PrettyTable(["Name"])
-        for name in groups:
+        for name in profiles:
             x.add_row([name])
 
     print x
 
 
+def profile_tag_show(name):
+    """Show the tags on the profile."""
+    try:
+        profile = client.get_profile(name)
+    except KeyError:
+        print "Profile %s not found." % name
+        sys.exit(1)
+
+    for tag in profile.tags:
+        print tag
+
+
+def profile_tag_add(name, tag):
+    """
+    Add a tag to the profile.
+    :param name: Profile name
+    :param tag: Tag name
+    :return: None
+    """
+    try:
+        profile = client.get_profile(name)
+    except KeyError:
+        print "Profile %s not found." % name
+        sys.exit(1)
+
+    profile.tags.add(tag)
+    client.profile_update_tags(profile)
+    print "Tag %s added to profile %s" % (tag, name)
+
+
+def profile_tag_remove(name, tag):
+    """
+    Remove a tag from the profile.
+    :param name: Profile name
+    :param tag: Tag name
+    :return: None
+    """
+    try:
+        profile = client.get_profile(name)
+    except KeyError:
+        print "Profile %s not found." % name
+        sys.exit(1)
+
+    try:
+        profile.tags.remove(tag)
+    except KeyError:
+        print "Tag %s is not on profile %s" % (tag, name)
+        sys.exit(1)
+    client.profile_update_tags(profile)
+    print "Tag %s removed from profile %s" % (tag, name)
+
+
+def profile_rule_show(name, human_readable=False):
+    """Show the rules on the profile."""
+    try:
+        profile = client.get_profile(name)
+    except KeyError:
+        print "Profile %s not found." % name
+        sys.exit(1)
+
+    if human_readable:
+        print "Inbound rules:"
+        i = 1
+        for rule in profile.rules.inbound_rules:
```

Comment:
Use enumerate rather than i=1 and i +=1
https://docs.python.org/2/library/functions.html#enumerate